### PR TITLE
Add docker image and go modules

### PR DESCRIPTION
**Pull request by julienduchesne**
go mod simplifies the build and the Dockerfile is useful since it's very likely that this would be used in Kubernetes

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM golang:1.16 as builder
+
+WORKDIR /app
+
+COPY . .
+
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build .
+
+FROM alpine:3.13
```
Comment (kbudde) [inline Dockerfile]: I know alpine image is quite small but I suspect it is not needed:
```
RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -installsuffix 'static' . 

FROM scratch
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+COPY --from=builder /app/alertmanager-webhook-logger /app/alertmanager-webhook-logger
+
```
Comment (kbudde) [inline Dockerfile:21]: You should not use root.
```
RUN mkdir /etc && \
    echo 'nobody:x:65534:65534:nobody:/:' > /etc/passwd && \
    echo 'nobody:x:65534:' > /etc/group

# Perform any further action as an unprivileged user.
USER nobody:nobody
```

**Review by kbudde — CHANGES_REQUESTED** [review]
I think dropping root priviliges is a must have.

Personally I would prefer a static build with image "scratch" as final.

**Comment by YvesMueller-TomTom** [comment]
This PR looks quite good. Could you please use an unprivileged user in the docker container as suggest by @kbudde ? I am fine with the Alpine image.

**Comment by julienduchesne** [comment]
Thanks for the reviews @yvesmueller-tomtom and @kbudde, and sorry for the delay. I switched the image to use scratch and an unprivileged user!

**Review by kbudde — APPROVED** [review]
LGTM

**Pull request merged**